httpx/client.py

```diff
@@ -217,6 +219,11 @@ def _get_auth_middleware(
         if callable(auth):
             return CustomAuthMiddleware(auth=auth)
 
+        if isinstance(auth, HTTPDigestAuth):
+            return HTTPDigestAuthMiddleware(
+                username=auth.username, password=auth.password
+            )
```

Comment:
With my "support `auth=<some middleware instance>`" suggest, this would look more like...

```python
if isinstance(auth, tuple):
    return BasicAuth(username=auth[0], password=auth[1])

elif isinstance(auth, BaseMiddleware):
    return auth

elif callable(auth):
    return CustomAuth(auth=auth)

else:
    # Bail!
```